### PR TITLE
test(aws_sdk): AWS SDK smoke tests

### PR DESCRIPTION
**Pull request by dnys1**
Adds an AWS SDK for smoke testing and smoke tests for basic services (API Gateway, DynamoDB, S3) to cover the supported AWS protocols. Uses the localstack project (https://github.com/localstack/localstack) in CI to verify behavior (as opposed to needing AWS credentials and running against a real backend).

### CODE REVIEW

File: .github/workflows/smoke_test.yml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+      - stable
+      - next
+    paths:
+      - 'packages/aws_sdk/**/*.dart'
+      - 'packages/smithy/**/*.dart'
+  pull_request:
+    paths:
+      - 'packages/aws_sdk/**/*.dart'
+      - 'packages/smithy/**/*.dart'
+  schedule:
```
Comment (ragingsquirrel3) [inline .github/workflows/smoke_test.yml:15]: nit: could this have a little comment saying when this runs for ppl (like me) who have to google cron syntax for this?

**Review by HuiSF — APPROVED** [review]

File: .github/workflows/smoke_test.yml [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    shell: bash
+permissions: read-all
+
+jobs:
+  smoke_test:
+    name: Smoke Test
+    runs-on: ubuntu-latest
+    steps:
+      - name: Git Checkout
+        uses: actions/checkout@2541b1294d2704b0964813337f33b291d3f8596b
```
Comment (ragingsquirrel3) [inline .github/workflows/smoke_test.yml:28]: nit: for these refs do you mind adding a comment saying which version number it is?

**Review by ragingsquirrel3 — APPROVED** [review]
approved w a few nits, thanks

File: packages/aws_sdk/smoke_test/test/s3_test.dart [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+import 'package:aws_common/aws_common.dart';
+import 'package:aws_signature_v4/aws_signature_v4.dart';
+import 'package:smithy_aws/smithy_aws.dart';
+import 'package:smoke_test/src/sdk/s3.dart';
+import 'package:test/test.dart';
+
+import 'common.dart';
+
+void main() {
+  group('S3', () {
```
Comment (HuiSF) [inline packages/aws_sdk/smoke_test/test/s3_test.dart:27]: should also add tests for APIs related to multipart upload?

Comment (dnys1) [reply packages/aws_sdk/smoke_test/test/s3_test.dart:27]: We can keep expanding it in future PRs. This was just to get a baseline.

**Pull request merged**